Create broadlink

roiff opened the pull request:
## Description:
Support broadlink RM device
The `broadlink` platform can create covers that combine components and provides the ability to run scripts or invoke services for each of the open, close, stop, and position commands of a cover.
Set the travel time to achieve precise control

## Example entry for `configuration.yaml` (if applicable):
```yaml
cover:
  - platform: broadlink
    host: 192.168.1.2
    mac: 'B4:43:0D:D5:25:CD'
    covers:
        bedroom_cover:
          travel_time: 9
          position_sensor: binary_sensor.door_window_sensor
          name: "bedroom_cover"
          command_open: 'sgY0AAwfIQoMHwwfCyAMHyAKIAoMHyAKDB8MHwwfDB8gCyAKDB8MHwsfCx8MHwwfIAohCgwAAUMAAAAA'
          command_close: 'sno0ACALIAsLIB8LCyALIAsgCyAgCyALHwsgCwsgCyALIAsgCyALIAsAAUQLICALCyALIAsgCyAAAAAA'
          command_stop: 'src0AB8LIAsLIAshCiELIAoAAUUKIR8LCyELIAsgCyAgDB8MCyAfDAohCyAKIAsgHwwfCwsgCyEAAAAA'

        living_cover:
          name: "living_cover"
          command_open: 'sgY0AAwfIQoMHwwfCyAMHyAKIAoMHyAKDB8MHwwfDB8gCyAKDB8MHwsfCx8MHwwfIAohCgwAAUMAAAAA'
          command_close: 'sno0ACALIAsLIB8LCyALIAsgCyAgCyALHwsgCwsgCyALIAsgCyALIAsAAUQLICALCyALIAsgCyAAAAAA'
          command_stop: 'src0AB8LIAsLIAshCiELIAoAAUUKIR8LCyELIAsgCyAgDB8MCyAfDAohCyAKIAsgHwwfCwsgCyEAAAAA'

        garage_door:
          name: "garage_door"
           position_sensor: binary_sensor.door_window_sensor
          command_open: 'sgY0AAwfIQoMHwwfCyAMHyAKIAoMHyAKDB8MHwwfDB8gCyAKDB8MHwsfCx8MHwwfIAohCgwAAUMAAAAA'
          command_close: 'sno0ACALIAsLIB8LCyALIAsgCyAgCyALHwsgCwsgCyALIAsgCyALIAsAAUQLICALCyALIAsgCyAAAAAA'
```

roiff committed:
Create broadlink

Danielhiversen commented:
Why not use https://home-assistant.io/components/cover.template/ ?

roiff commented:
@Danielhiversen 
Easy setting
Do not need to pass broadlink swich 
Support travel time
Direct support HK windows covers
template configuration trouble
Especially when there are multiple curtains

roiff commented:
If you do not use the plug-in
Still have to set up a script to run better

tinloaf commented:
I very much think we should not have protocol-level codes as part of the configuration…

roiff committed:
Update broadlink

Reconstruction of some logic, to be more accurate
Support for garage doors and windows without travel
Optimized door and window sensing logic

roiff commented:
I think this will be very helpful for the RF controlled curtains and the garage
At least at my side, many people use it

Danielhiversen commented:
I still think it is better to use the template cover

roiff commented:
@Danielhiversen 
Because most RF curtains do not have state feedback
Replace the state with the timeline
Using templates can get tricky
For curtains with state sensing and feedback, I also highly recommend templates

roiff commented:
There is another reason
HK does not support curtain types with a pause button if there is no timeline(position status)...
This is helpful for ease of use and experience

Danielhiversen commented:
@balloob : Can you decide if we want this component?

balloob commented:
Users should indeed use the `cover.template`. If we would allow this, we would have to allow thousands of semi-integrations.
